### ISSUE TITLE
Scale Grading With Multiple Parts Causes Incorrect assignment of Grade to Score

### ISSUE THREAD

**Issue by DownesAyr** [turnitin/moodle-mod_turnitintooltwo#243]
Scale grading allocates incorrect grade when using multiple parts within a single Turnitin 2 assignment. For example I used Scale: pass, fail, remediate and a maximum mark of 3. This way if an assignment is only using one part a mark of 1 constitutes a fail, 2 constitutes a remediate and 3 a pass. However the moment a second part is added to the activity a 2 changes to a fail grade. 

See below two images. The first with only one part and showing correct grading, the second with multiple parts showing incorrect grading.

### WORKING SCALE GRADE WITH 1 PART
![scale grading normal](https://cloud.githubusercontent.com/assets/22812156/21262042/fb7d2ef8-c386-11e6-982d-8a477de4cc8d.PNG)

### BUGGY SCALE GRADE WITH MULTIPLE PARTS
![scale grading issue](https://cloud.githubusercontent.com/assets/22812156/21262045/fedfcc68-c386-11e6-87e8-29428edc5ffc.PNG)

I am using Moodle 3.1.3+ and Turnitin 2 (2016091401) version 2.6+

**Comment by jmcgettrick** [comment]
Hi @DownesAyr, this will be because it's including the other parts despite not being graded in the overall grade e.g. 2 + 0 + 0 would be 2/9 therefore a fail.

**Comment by DownesAyr** [comment]
Thanks for your quick reply. 

So as far as I can see the only way to get the response I want is to go in and change each part to match. For example: say a student has only submitted part 1 - you want their grade to be **remediate** so I would put 2/3 in every part even before the student submitted anything to parts 2 or 3. Just so the overall grade balances out.

From what it seems it would maybe make more sense to have a grade for each part and have the overall grade as separate field that need only make sense after all parts are completed...

**Comment by jmcgettrick** [comment]
Yes @DownesAyr that does make more sense, we'll get this looked at. The only problem I can see is if a user doesn't submit to a part then they don't get an overall grade.

**Comment by DownesAyr** [comment]
That would be great if you could add in those aspects. Regarding the overall grade I don't see a problem as long as there is a column that exists for the part grade. The overall grade could still function as it does currently to save on you having to make too many code changes.

**Comment by jack-tii** [comment]
Hi, I know this is an old request, but if this is still a feature you're interested in please open a ticket with support
